core/server/common/src/main/java/alluxio/master/journal/AsyncJournalWriter.java

```diff
@@ -302,28 +292,24 @@ private void doFlush() {
         }
 
         // Notify tickets that have been served to wake up.
-        try (LockResource lr = new LockResource(mTicketLock)) {
-          ListIterator<FlushTicket> ticketIterator = mTicketList.listIterator();
-          while (ticketIterator.hasNext()) {
-            FlushTicket ticket = ticketIterator.next();
-            if (ticket.getTargetCounter() <= mFlushCounter.get()) {
-              ticket.setCompleted();
-              ticketIterator.remove();
-            }
+        Iterator<FlushTicket> ticketIterator = mTicketSet.iterator();
+        while (ticketIterator.hasNext()) {
```

Comment:
will this iterator be correct w.r.t. the concurrency?

Reply:
Whether doFlush should enter or not, is controlled by a semaphore. So as long as there is an RPC that needs a flush, doFlush will spin.

Because of that even if an active iteration of doFlush misses a ticket due to concurrency, it'll iterate one more because of the permit given by the ticket holder. And that guarantees the correctness.